pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -299,6 +299,23 @@ type Trigger struct {
 	// Retry strategy, defaults to no retry
 	// +optional
 	RetryStrategy *apicommon.Backoff `json:"retryStrategy,omitempty" protobuf:"bytes,4,opt,name=retryStrategy"`
+	// Rate limit, default unit is Second
+	// +optional
+	RateLimit *RateLimit `json:"rateLimit,omitempty" protobuf:"bytes,5,opt,name=rateLimit"`
+}
+
+type RateLimiteUnit string
+
+const (
+	Second RateLimiteUnit = "Second"
+	Minute RateLimiteUnit = "Minute"
+	Hour   RateLimiteUnit = "Hour"
+)
+
+type RateLimit struct {
+	// Defaults to Second
+	Unit            RateLimiteUnit `json:"unit,omitempty" protobuf:"bytes,1,opt,name=unit"`
```

Comment:
can't you use Duration here?

Reply:
Using Duration like below seems to be weird:

```yaml
spec:
  triggers:
    - rateLimit:
        unit: 5s
        requests: 20
```

Referencing to `Istio` and `ambassador`, both of them use `unit`:

Istio:
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: ratelimit-config
data:
  config.yaml: |
    domain: productpage-ratelimit
    descriptors:
      - key: PATH
        value: "/productpage"
        rate_limit:
          unit: minute
          requests_per_unit: 1
      - key: PATH
        rate_limit:
          unit: minute
          requests_per_unit: 100
```

Ambassador:
```yaml
apiVersion: getambassador.io/v2
kind: RateLimit
metadata:
  name: backend-rate-limit
spec:
  domain: ambassador
  limits:
   - pattern: [{generic_key: backend}]
     rate: 3
     unit: minute
```